src/tables.jl

```diff
@@ -115,3 +115,86 @@ end
 
 #     end
 # end
+
+"""
+    AxisKeys.populate!(A, table, value; force=false, quiet=false)
+
+Populate `A` with the contents of the `value` column in a provided `table`.
+The `table` must contain columns corresponding to the keys in `A` and support row iteration.
+If the keys in `A` do not uniquely identify rows in the `table` then an
+`ArgumentError` is throw. If `force` is true then the duplicate (non-unique) entries will be
+overwritten. Setting `quiet` to true will avoid any non-unique warnings.
+"""
+function populate!(A, table, value::Symbol; force=false, quiet=false)
+    # Use a BitArray mask to detect duplicates and error instead of overwriting.
+    mask = falses(size(A))
+    overwritten = false
+
+    for r in Tables.rows(table)
+        vals = Tuple(Tables.getcolumn(r, c) for c in dimnames(A))
+        inds = map(findindex, vals, axiskeys(A))
+
+        # Handle duplicate error if applicable
+        if mask[inds...]
+            if force
+                overwritten = true
+            else
+                throw(ArgumentError("Key $vals is not unique"))
+            end
+        end
+
+        # Set or mask marking that we've set this index
+        setindex!(mask, true, inds...)
+
+        # Insert our value into the data array
+        setindex!(A, Tables.getcolumn(r, value), inds...)
+    end
+
+    if overwritten && !quiet
+        @warn "Columns $(dimnames(A)) do not uniquely identify rows in the table"
+    end
+
+    return A
+end
+
+"""
+    wrapdims(table, [Type,] value, keys...; default=undef, sort=false, force=false, quiet=false)
+
+Construct a KeyedArray/NamedDimsArray from a `table` supporting column and row.
```

Comment:
Yeah, I think I was mimicing an existing `wrapdims` method that has a `Type` as the second argument... but I guess that's used for wrapping the keys rather than explicitly specifying the return type.

Reply:
I've moved it to be the first argument and explicitly stated that the `table` should implement `Tables.columns` and `Tables.rows`.